File: benchmarks/java/integer_boolean/Clojure_Big_Integer_From_Java/Equivalent.java
import de.hhu.stups.btypes.BInteger;
import de.hhu.stups.btypes.BBoolean;

public class Equivalent {






    private BInteger counter;

    public Equivalent() {
        counter = (BInteger) new BInteger("0");
    }

    public void simulate() {
        while((counter.less(new BInteger("5000000")).equivalent(new BInteger("1").equal(new BInteger("1")))).booleanValue()) {
            counter = (BInteger) counter.plus(new BInteger("1"));
        }
    }

    public static void main(String[] args) {
        Equivalent exec = new Equivalent();
        long start = System.nanoTime();
        exec.simulate();
        long end = System.nanoTime();
        System.out.println(exec.getClass().toString() + " Execution: " + (end - start));
    }

}

File: benchmarks/cpp/integer_boolean/big_integer/Equivalent.cpp
#include <iostream>
#include <string>
#include "BUtils.cpp"
#include "BInteger.cpp"
#include "BBoolean.cpp"

#ifndef Equivalent_H
#define Equivalent_H

using namespace std;

class Equivalent {



    private:



        BInteger counter;

    public:

        Equivalent() {
            counter = (BInteger("0"));
        }

        void simulate() {
            while((counter.less((BInteger("5000000"))).equivalent((BInteger("1")).equal((BInteger("1"))))).booleanValue()) {
                counter = counter.plus((BInteger("1")));
            }
        }

};
int main() {
    clock_t start,finish;
    double time;
    Equivalent exec;
    start = clock();
    exec.simulate();
    finish = clock();
    time = (double(finish)-double(start))/CLOCKS_PER_SEC;
    printf("%f\n", time);
    return 0;
}
#endif

